From f9aa5f662cb941851e78f3875328dd39fcfcf58e Mon Sep 17 00:00:00 2001
From: cnauroth <cnauroth@apache.org>
Date: Fri, 9 Oct 2015 14:27:13 -0700
Subject: [PATCH 1592/2848] HDFS-9085. Show renewer information in
 DelegationTokenIdentifier#toString. Contributed
 by Zhihai Xu.

(cherry picked from commit 4f6e842ba936f4a068168b518eea80bb6dd02d85)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenIdentifier.java
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: If08b8adc2c732c3cb111428ca90dbf46abed732e
---
 .../delegation/DelegationTokenIdentifier.java      |    3 ++-
 .../hadoop/hdfs/security/TestDelegationToken.java  |    9 +++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenIdentifier.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenIdentifier.java
index 38ac687..83b6173 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenIdentifier.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenIdentifier.java
@@ -61,7 +61,8 @@ public Text getKind() {
   @Override
   public String toString() {
     return getKind() + " token " + getSequenceNumber()
-        + " for " + getUser().getShortUserName();
+        + " for " + getUser().getShortUserName() + " with renewer " +
+        getRenewer();
   }
 
   /** @return a string representation of the token */
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/security/TestDelegationToken.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/security/TestDelegationToken.java
index f98a1f6..1bf4429 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/security/TestDelegationToken.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/security/TestDelegationToken.java
@@ -315,4 +315,13 @@ public Object run() throws Exception {
           }
         });
   }
+
+  @Test
+  public void testDelegationTokenIdentifierToString() throws Exception {
+    DelegationTokenIdentifier dtId = new DelegationTokenIdentifier(new Text(
+        "SomeUser"), new Text("JobTracker"), null);
+    Assert.assertEquals("HDFS_DELEGATION_TOKEN token 0" +
+        " for SomeUser with renewer JobTracker",
+        dtId.toString());
+  }
 }
-- 
1.7.9.5

